### PR TITLE
[FIX] account: wrong destination account id in internal transfer

### PR DESCRIPTION
**Pull request by agr-odoo**
Setup at least 2 bank accounts [ACC1] and [ACC2]
Setup a Journal [BNK1] with [ACC1]
Setup a second journal [BNK2] with [ACC2]
Create an Internal Transfer from [BNK1] to [BNK2]
Save and confirm

partner_bank_id will be choosen from the Journal, for inbound
payment, ([ACC1]) or it from one of the company bank accounts in
outbound payment, so either [ACC1] or [ACC2]
This is wrong as [ACC2] should be used because defined on [BNK2]

opw-2794081

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/account/models/account_payment.py [hunk 1]
```diff
@@ -380,14 +380,17 @@ def _compute_amount_signed(self):
...
             if pay.payment_type == 'inbound':
                 pay.available_partner_bank_ids = pay.journal_id.bank_account_id
             else:
-                pay.available_partner_bank_ids = pay.partner_id.bank_ids\
-                        .filtered(lambda x: x.company_id.id in (False, pay.company_id.id))._origin
+                if pay.is_internal_transfer:
+                    pay.available_partner_bank_ids = pay.destination_journal_id.bank_account_id
+                else:
+                    pay.available_partner_bank_ids = pay.partner_id.bank_ids\
+                            .filtered(lambda x: x.company_id.id in (False, pay.company_id.id))._origin
```
Comment (Levizar) [inline addons/account/models/account_payment.py:393]:
```suggestion
            elif pay.is_internal_transfer:
                pay.available_partner_bank_ids = pay.destination_journal_id.bank_account_id
            else:
                pay.available_partner_bank_ids = pay.partner_id.bank_ids\
                        .filtered(lambda x: x.company_id.id in (False, pay.company_id.id))._origin
```

File: addons/account/models/account_payment.py [hunk 2]
```diff
@@ -380,14 +380,17 @@ def _compute_amount_signed(self):
             else:
                 payment.amount_signed = payment.amount
 
-    @api.depends('partner_id', 'company_id', 'payment_type')
+    @api.depends('partner_id', 'company_id', 'payment_type', 'destination_journal_id', 'is_internal_transfer')
     def _compute_available_partner_bank_ids(self):
         for pay in self:
             if pay.payment_type == 'inbound':
                 pay.available_partner_bank_ids = pay.journal_id.bank_account_id
             else:
```
Comment (Levizar) [inline addons/account/models/account_payment.py:388]:
```suggestion
```

**Review by Levizar — CHANGES_REQUESTED** [review]
At first sight, it seems completely legit to me.

**Comment by agr-odoo** [comment]
@Levizar Done, thanks

**Review by Levizar — CHANGES_REQUESTED** [review]

File: addons/account/models/account_payment.py [hunk 3]
```diff
@@ -380,11 +380,13 @@ def _compute_amount_signed(self):
...
                 payment.amount_signed = payment.amount
 
-    @api.depends('partner_id', 'company_id', 'payment_type')
+    @api.depends('partner_id', 'company_id', 'payment_type', 'destination_journal_id', 'is_internal_transfer')
     def _compute_available_partner_bank_ids(self):
         for pay in self:
             if pay.payment_type == 'inbound':
                 pay.available_partner_bank_ids = pay.journal_id.bank_account_id
+            elif  pay.is_internal_transfer:
```
[the remark this answers is not in the document]
Comment (agr-odoo) [reply addons/account/models/account_payment.py]: how I didn't notice this... :thinking: 
fixed, thanks for noticing :+1:

**Review by Levizar — APPROVED** [review]
LGTM

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**